addons/l10n_ch/models/account_invoice.py

```diff
@@ -33,6 +34,14 @@ def _compute_l10n_ch_qr_is_valid(self):
                 )
             )
 
+    @api.depends('company_id', 'state')
+    def _compute_display_qr_code(self):
+        # Extends account
+        super()._compute_display_qr_code()
+        moves_ch = self.filtered(lambda m: m.company_id.account_fiscal_country_id.code == 'CH')
+        for move in moves_ch:
+            move.display_qr_code = move.state != 'draft'
```

Comment:
I'm guessing this shouldn't be the only condition ? Won't this bypass every other QR verification ? Won't this affect qr codes emitted from CH yet not CH qr ?   
As we talked about earlier, I feel like we're mistaking the "classic QR" (sepa etc) behavior with the l10n_ch one. I'll have a better look.

Reply:
benj:
iban qui commence par ch
digit [5:]: between 30000 et 31999